### PR TITLE
libct.Start: fix locking, do not allow a second container init

### PR DESCRIPTION
**Pull request by kolyshkin**
By definition, every container has only 1 init (i.e. PID 1) process.

Apparently, libcontainer API supported running more than 1 "init", and
two tests mistakenly used it. Of course, the second "init" was not
PID 1, but it was started like init and, I guess, some of the Container fields
were set to wrong values.

Let's not allow that, erroring out if we already have init running.

Fix two cases in libct/int which ran two inits inside a container.
Also, fix a locking issue and remove some code that's not needed.

### CODE REVIEW

**Comment by kolyshkin** [comment]
@opencontainers/runc-maintainers PTAL

**Review by AkihiroSuda — APPROVED** [review]

**Comment by kolyshkin** [comment]
@lifubang ptal

**Comment by lifubang** [comment]
> fix a locking issue

SGTM, but could you give some details for this issue in commit [3b5376f](https://github.com/opencontainers/runc/pull/4271/commits/3b5376f583baa14b51d96a3f5976c79ab7177368)? I think it's a refactor, but not a fix?

File: libcontainer/container_linux.go
Comment (lifubang) [reply libcontainer/container_linux.go:315]: My suggestion change is unneeded Once #4319 merged.

**Comment by kolyshkin** [comment]
> > fix a locking issue
> 
> SGTM, but could you give some details for this issue in commit [3b5376f](https://github.com/opencontainers/runc/pull/4271/commits/3b5376f583baa14b51d96a3f5976c79ab7177368)? I think it's a refactor, but not a fix?

Maybe the commit message is too long and vague. Here's an except about the main issue: "part of c.Run executing without the lock". I did not get too deep inside why this locking is needed, but if it is needed, it is a bug to execute Run in parallel with, say, Start. This is what's being fixed.

**Comment by lifubang** [comment]
> Here's an except about the main issue: "part of c.Run executing without the lock".

I understand now, thanks your explanation.

**Review by lifubang — APPROVED** [review]

**Pull request merged**
